src/Microsoft.Diagnostics.Monitoring.WebApi/Exceptions/ExceptionsConfiguration.cs

```diff
@@ -0,0 +1,134 @@
+            {
+                if (evaluateFilterList(configuration, topFrame))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        internal bool ShouldInclude(IExceptionInstance exception)
+        {
+            Func<ExceptionConfiguration, CallStackFrame, bool> evaluateFilterList = (configuration, topFrame) =>
+            {
+                bool include = true;
+                if (topFrame != null)
+                {
+                    CompareIncludeValues(configuration.MethodName, topFrame.MethodName, ref include);
+                    CompareIncludeValues(configuration.ModuleName, topFrame.ModuleName, ref include);
+                    CompareIncludeValues(configuration.ClassName, topFrame.ClassName, ref include);
+                }
+
+                CompareIncludeValues(configuration.ExceptionType, exception.TypeName, ref include);
+
+                return include;
+            };
+
+            return CheckConfiguration(exception, Include, evaluateFilterList);
+        }
+
+        internal bool ShouldExclude(IExceptionInstance exception)
+        {
+            Func<ExceptionConfiguration, CallStackFrame, bool> evaluateFilterList = (configuration, topFrame) =>
+            {
+                bool? exclude = null;
+                if (topFrame != null)
+                {
+                    CompareExcludeValues(configuration.MethodName, topFrame.MethodName, ref exclude);
+                    CompareExcludeValues(configuration.ModuleName, topFrame.ModuleName, ref exclude);
+                    CompareExcludeValues(configuration.ClassName, topFrame.ClassName, ref exclude);
+                    CompareExcludeValues(configuration.ExceptionType, exception.TypeName, ref exclude);
+                }
+                else
+                {
+                    CompareExcludeValues(configuration.ExceptionType, exception.TypeName, ref exclude);
+                }
+
+                return exclude ?? false;
+            };
+
+            return CheckConfiguration(exception, Exclude, evaluateFilterList);
+        }
+
+        private static void CompareIncludeValues(string configurationValue, string actualValue, ref bool include)
+        {
+            if (include && !string.IsNullOrEmpty(configurationValue))
+            {
+                include = CompareValues(configurationValue, actualValue);
+            }
+        }
+
+        private static void CompareExcludeValues(string configurationValue, string actualValue, ref bool? exclude)
+        {
+            if (exclude != false && !string.IsNullOrEmpty(configurationValue))
+            {
+                exclude = CompareValues(configurationValue, actualValue);
+            }
+        }
+
+        private static bool CompareValues(string configurationValue, string actualValue)
+        {
+            return actualValue.Contains(configurationValue, StringComparison.CurrentCultureIgnoreCase);
```

Comment:
In the original discussion for this, there was some debate over what level of fuzzy-matching we wanted - whether we wanted to check for an exact match unless a wildcard was provided. I did prototype it this way (and it'd be easy to swap it back), but this to me feels more user-friendly - when in doubt, I'd rather over-show than under-show and have people try to make sure they've got their names exactly right...e.g. I want to be able to say "InvalidOperationException" instead of needing to do "System.InvalidOperationException"